cli/tests/ld-floxlib.bats

```diff
@@ -0,0 +1,132 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test that LD_AUDIT and ld-floxlib.so works as expected on Linux only.
+#
+# This test loads up a flox environment containing the following packages:
+# * gcc (to be able to compile the test program)
+# * a pinned version of glibc 2.34 from the past
+# * patchelf (to modify ELF binaries)
+# * nix (a package that is guaranteed to be not available in FHS lib)
+# * curl, libarchive (runtime libraries required by libnixmain.so)
+#
+# It then activates the env to perform two distinct tests:
+# 1: load libraries found in $FLOX_ENV_LIBS last
+#   - compile the get-glibc-version program (using $FLOX_ENV/{include,lib})
+#   - remove its custom RUNPATH and ld interpreter so that it will use the
+#     "system" libc
+#   - run it having cleared the environment (with `env -i`) and observe the
+#     default glibc version, confirm this does NOT match the pinned version
+#   - repeat with LD_AUDIT defined and confirm that the version again does
+#     not change
+#   - repeat with LD_LIBRARY_PATH=$FLOX_ENV_LIBS and confirm that this rolls
+#     back glibc to the version installed to the environment
+# 2: confirm LD_AUDIT can find missing libraries
+#   - compile the get-nix-version program
+#   - observe that it can run the compiled program
+#   - unset LD_AUDIT and confirm it cannot run the program
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+
+# bats file_tags=end2end,ld-floxlib
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup
+}
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-${BATS_TEST_NUMBER?}"
+  export PROJECT_NAME="${PROJECT_DIR##*/}"
+  rm -rf "$PROJECT_DIR"
+  mkdir -p "$PROJECT_DIR"
+  cp ./ld-floxlib/* "$PROJECT_DIR"
+  pushd "$PROJECT_DIR" > /dev/null || return
+  _PKGDB_GA_REGISTRY_REF_OR_REV="${PKGDB_NIXPKGS_REV_OLDER?}" \
+    "$FLOX_BIN" init
+}
+
+project_teardown() {
+  popd > /dev/null || return
+  # rm -rf "${PROJECT_DIR?}"
+  unset PROJECT_DIR
+  unset PROJECT_NAME
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup
+  project_setup
+}
+teardown() {
+  project_teardown
+  common_test_teardown
+}
+
+# ---------------------------------------------------------------------------- #
+#
+@test "test ld-floxlib.so on Linux only" {
+  if [ $(uname -s) != "Linux" ]; then
+    skip "not Linux"
+  fi
+
+  # Note:
+  # - installing old versions of nix (2.10.3) and glibc (2.34) for use in tests
+  # - installing curl and libarchive because those packages provide libraries
+  #   that are runtime dependencies of libnixmain.so
```

Comment:
In discussion with Matthew we concluded removing that lock wouldn't help because another test would just recreate it. He instead suggested we run `flox update` to cement the choice of nixpkgs revision.